src/App.tsx

```diff
@@ -1,14 +1,107 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
+
 import { Loader } from './components/Loader';
+import { debounce } from './_utils/debounce_generic';
 
 export const App: React.FC = () => {
+  const [loading, setLoading] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>(todos);
+  const [currentSortOption, setSortOption] = useState('all');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [isTodoModalVisibile, setTodoModalVisibility] = useState(false);
+
+  const loadTodos = async () => {
+    setLoading(true);
+
+    try {
+      const loadedTodos = await getTodos();
+
+      setTodos(loadedTodos);
+    } catch (error) {
+      setLoading(false);
+    } finally {
+      setLoading(false);
+    }
+  };
```

Comment:
```suggestion
  const loadTodos = async () => {
    setLoading(true);

    try {
      const loadedTodos = await getTodos();

      setTodos(loadedTodos);
      setFilteredTodos(loadedTodos);
    } catch (error) {
      setLoading(false);
    } finally {
      setLoading(false);
    }
  };
```

Reply:
you can set initially filtered todos when they're loaded so you won't need additional useEffect